web/gds-admin-ui/src/helpers/api/apiCore.js

```diff
@@ -186,11 +199,29 @@ class APICore {
 
         const decodedAccessToken = jwtDecode(user.access_token);
         const currentTime = Date.now() / 1000;
-        if (decodedAccessToken.exp < currentTime) {
+        const csrfToken = getCookie('csrf_token');
+
+        if (currentTime < decodedAccessToken.exp && currentTime > decodedAccessToken.nbf) {
+            if (isValidRefreshToken(user.refresh_token)) {
```

Comment:
The `refresh_token` is stored with the user - I didn't see where this was stored? Is the user also stored in the session state?

Reply:
The user object `( access_token & refresh token )` is stored in the session